optuna/study.py

```diff
@@ -276,6 +276,18 @@ def set_system_attr(self, key, value):
 
         self._storage.set_study_system_attr(self.study_id, key, value)
 
+    def set_trial_user_attr(self, trial_id, key, value):
```

Comment:
It seems somewhat strange that `Study` class provides `sed_trial_{user,system}_attr` methods, because `Trial` class already has `set_{user,system}_attr` methods for the same purposes.
Ideally, in the future, we want to replace `FrozenTrial` with `Trial` (or something similar class) to provide more consistent API to users.
In the meantime, we would like to set trial attributes via the existing methods of `_storage` (instead of adding new methods to `Study` class).
What do you think?

Reply:
I felt the same but I couldn't find the good solution about this. I think your suggestion is better than adding methods that might be breaking changes. So I'll use `_storage` in this pr.